ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/CSJ.hs

```diff
@@ -124,17 +124,19 @@ prop_CSJ happy synchronized =
           receivedHeadersAtMostOnceFromHonestPeers
     )
   where
-    genDuplicatedHonestSchedule :: GenesisTest TestBlock () -> Gen (PeersSchedule TestBlock)
+    genDuplicatedHonestSchedule :: GenesisTest TestBlock () -> Gen (PointSchedule TestBlock)
     genDuplicatedHonestSchedule gt@GenesisTest {gtExtraHonestPeers} = do
-      Peers {honestPeers, adversarialPeers} <- genUniformSchedulePoints gt
-      pure $
-        Peers.unionWithKey
-          (\_ _ _ -> error "should not happen")
-          ( peers'
-              (replicate (fromIntegral gtExtraHonestPeers + 1) (getHonestPeer honestPeers))
-              []
-          )
-          (Peers Map.empty adversarialPeers)
+      ps <- genUniformSchedulePoints gt
+      pure $ ps {
```

Comment:
In this specific case, it was on purpose because I expected the duplication of the honest schedule to not have any impact on the other fields. I'm happy introducing a pattern destruction, though!